models/MARBL_column/model_mod.f90

```diff
@@ -0,0 +1,638 @@
+                                 NF90_MAX_NAME, nc_open_file_readonly, &
+                                 nc_get_variable, nc_get_variable_size, nc_close_file
+
+use state_structure_mod, only : add_domain, get_domain_size, get_model_variable_indices, &
+                                get_varid_from_kind, get_dart_vector_index, &
+                                get_num_domains
+
+use obs_kind_mod, only : get_index_for_quantity, QTY_U_CURRENT_COMPONENT, &
+                         QTY_V_CURRENT_COMPONENT, QTY_DRY_LAND, &
+                         QTY_LAYER_THICKNESS
+
+use distributed_state_mod, only: get_state
+
+use ensemble_manager_mod, only : ensemble_type
+
+! These routines are passed through from default_model_mod.
+! To write model specific versions of these routines
+! remove the routine from this use statement and add your code to
+! this the file.
+use default_model_mod, only : pert_model_copies, write_model_time, &
+                              init_time => fail_init_time, &
+                              init_conditions => fail_init_conditions, &
+                              convert_vertical_obs, convert_vertical_state, adv_1step
+
+implicit none
+private
+
+! routines required by DART code - will be called from filter and other
+! DART executables. 
+public :: get_model_size,         &
+          get_state_meta_data,    &
+          model_interpolate,      &
+          end_model,              &
+          static_init_model,      &
+          nc_write_model_atts,    &
+          get_close_obs,          &
+          get_close_state,        &
+          pert_model_copies,      &
+          convert_vertical_obs,   &
+          convert_vertical_state, &
+          read_model_time,        &
+          adv_1step,              &
+          init_time,              &
+          init_conditions,        &
+          shortest_time_between_assimilations, &
+          write_model_time
+
+
+character(len=256), parameter :: source = "model_mod.f90"
+logical :: module_initialized = .false.
+integer :: state_dom_id  ! used to access the state structure
+integer :: param_dom_id  ! used to access MARBL internal parameters
+integer :: nfields ! number of fields in the state or parameter vector
+integer :: nz      ! the number of vertical layers
+integer :: model_size
+type(time_type) :: assimilation_time_step
+real(r8) :: geolon 
+real(r8) :: geolat 
+real(r8) :: basin_depth(2,2)
+
+! parameters to be used in specifying the DART internal state
+integer, parameter :: modelvar_table_height = 13
+integer, parameter :: modelvar_table_width = 5
+integer, parameter :: modelparams_table_height = 1
+integer, parameter :: modelparams_table_width = 5
+
+! defining the variables that will be read from the namelist
+character(len=256) :: template_file(2)
+character(len=256) :: ocean_geometry
+real(r8)           :: station_location(2)
+integer            :: time_step_days
+integer            :: time_step_seconds
+logical            :: estimate_params
```

Comment:
all the namelist variables need default values.  

otherwise you have the value set to whatever junk is in memory if the namelist value is not set in input.nml